deploy/lib/server_config.rb

```diff
@@ -1362,6 +1379,41 @@ def settings
 
 private
 
+  def filter_to_newer_files(files, options)
+    logger.info "Filtering to files which are newer locally than on the server"
+
+    if @server_version < 6
+      raise ExitException.new("Can only filter files on MarkLogic 6 and later")
+    end
+
+    uris = files.map { |f| xcc.build_target_uri(f, options) }
+    stamps_db = get_db_timestamps(uris)
+    stamps_local = files.map { |file_uri| File.mtime(file_uri).getgm.iso8601(5).tr('Z','') }
+
+    files_with_stamps = files.zip(stamps_local, stamps_db)
+
+    filtered = files_with_stamps.select do |file_uri, stamp_locally, stamp_in_db|
+
+      newer = (stamp_locally > stamp_in_db || stamp_in_db.strip.empty?)
+
+      if (!newer)
+        logger.debug "Ignoring #{file_uri} as server version is newer"
+      end
+
+      newer
+    end
+
+    filtered.map { |f, stamp1, stamp2| f}
+  end
+
+  def get_db_timestamps(uris)
+    uris_as_string = uris.map{|i| "\"#{i}\""}.join(",")
+    q = %Q{for $u in (#{uris_as_string}) return "" || xdmp:timestamp-to-wallclock(xdmp:document-timestamp($u))}
+
+    result = execute_query q, :db_name => @properties["ml.content-db"]
```

Comment:
I think this feature would be very much appreciated for deploy modules as well, at the least. For that it would be useful if it would be more dynamic..